### PR TITLE
refactor: odp graphql query

### PR DESCRIPTION
**Pull request by andrewleap-optimizely**
## Summary
- Modify odp graphql query to utilize graphql variables.

Allows the query itself to be separate from user provided data.

## Test plan
- all existing tests pass
## Ticket
- [FSSDK-8477](https://jira.sso.episerver.net/projects/FSSDK/issues/FSSDK-8477)

### CODE REVIEW

File: lib/optimizely/odp/odp_segments_api_manager.rb [hunk 1]
```diff
@@ -41,8 +41,13 @@ def fetch_segments(api_key, api_host, user_key, user_value, segments_to_check)
...
 
       payload = {
-        'query' => %'query {customer(#{user_key}: "#{user_value}")' \
-                   "{audiences(subset:#{segments_to_check || '[]'}) {edges {node {name state}}}}}"
+        query: 'query($vuid: String, $fs_user_id: String, $audiences: [String]) {' \
+               'customer(vuid: $vuid, fs_user_id: $fs_user_id) ' \
+               '{audiences(subset: $audiences) {edges {node {name state}}}}}',
+        variables: {
+          user_key => user_value.to_s,
+          audiences: segments_to_check || []
```
Comment (jaeopt) [inline lib/optimizely/odp/odp_segments_api_manager.rb:49]: Can you take a look at the format in csharp SDK? https://github.com/optimizely/csharp-sdk/blob/8f23e031c5b766ea8ba678d55114be3cf9c3a494/OptimizelySDK/Odp/OdpSegmentApiManager.cs#L152
This one is more clean in that we do not list all accepted query-ids. Let me know if you see any concern. It'll be good if we can use the same format for all SDKs.

File: lib/optimizely/odp/odp_config.rb [hunk 2]
```diff
@@ -46,7 +46,9 @@ def initialize(api_key = nil, api_host = nil, segments_to_check = [])
     def update(api_key = nil, api_host = nil, segments_to_check = [])
       updated = false
       @mutex.synchronize do
+        initial_state = @odp_state
         @odp_state = api_host.nil? || api_key.nil? ? ODP_CONFIG_STATE[:NOT_INTEGRATED] : ODP_CONFIG_STATE[:INTEGRATED]
+        updated = @odp_state != initial_state
```
Comment (jaeopt) [inline lib/optimizely/odp/odp_config.rb:51]: Not clear why this update again in addition to line 57? Does it make difference?

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good. A couple of clarifications.

**Comment by jaeopt** [comment]
One more suggestion. Can we clean up these names for consistency?
odp_events_api_manager -> odp_event_api_manager
odp_segments_api_manager -> odp_segment_api_manager

File: lib/optimizely/odp/odp_segments_api_manager.rb (hunk 1 above)
Comment (andrewleap-optimizely) [reply lib/optimizely/odp/odp_segments_api_manager.rb:49]: Yea I think that is the better solution, I'll update this one to match.

**Review by jaeopt — APPROVED** [review]
LGTM

**Review by zashraf1985 — APPROVED** [review]

**Pull request merged**